snippets/python-snippets.json

```diff
@@ -192,39 +192,63 @@
         ],
         "description": "Add onchange field"
     },
-    "Constrains on field": {
+    "SQL Constraints": {
+        "prefix": "oofconstrains",
+        "body": [
+            "_sql_constraints = [",
+            "\t(\"${1:field_name}_${2|check,unique|}\", \"${2}(${1})\", \"${3:Error message}\"),",
+            "]"
+        ],
+        "description": "Add SQL constraints to check or unique field"
+    },
+    "Python Constrains": {
         "prefix": "oofconstrains",
         "body": [
             "@api.constrains('${1:fieldname}')",
             "def _constrains_${1}(self):",
             "\t${0:pass}"
         ],
-        "description": "Add constrains on field"
+        "description": "Add a python constrains"
+    },
+    "Ensure one record": {
+        "prefix": "ooone",
+        "body": [
+            "self.ensure_one()"
+        ],
+        "description": "Ensure one record passing to this method"
+    },
+    "Super hook method": {
+        "prefix": "oosuper",
+        "body": [
+            "def ${1:method_name}(self, ${2:vals}):",
+            "\tres = super(${3:model_name}, self).${1:method_name}(${2:vals})",
+            "\t${0}",
+            "\treturn res"
+        ],
+        "description": "Add super hook method"
     },
     "Create method": {
         "prefix": "oocreate",
         "body": [
-            "@api.${1|model,model_create_multi|}",
+            "@api.model_create_multi",
             "def create(self, ${2:vals}):",
-            "\treturn super().create(${2:vals})"
+            "\treturn super(${1:model_name}, self).create(${2:vals})"
         ],
         "description": "Add create method"
     },
     "Write method": {
         "prefix": "oowrite",
         "body": [
-            "@api.multi",
             "def write(self, vals):",
-            "\treturn super().write(vals)"
```

Comment:
Can you revert this? Bcz no need to pass arguments while calling super from python3.

Reply:
done, thanks bro. I don't even know that 😆